ue5osc/osc_dispatcher.py

```diff
@@ -0,0 +1,63 @@
+from time import sleep
+from typing import Any, List
+
+from pythonosc.dispatcher import Dispatcher
+
+
+class OSCMessageReceiver:
+    def __init__(self):
+        self.values = None
+        self.dispatcher = Dispatcher()
+
+        # Map commands to the set_filter method
+        self.dispatcher.map("/location", self.handle_location)
+        self.dispatcher.map("/rotation", self.handle_rotation)
+        self.dispatcher.map("/project", self.handle_project)
+        self.dispatcher.set_default_handler(self.handle_invalid_command)
+
+    def handle_location(
+        self, address: str, *args: List[Any]
+    ) -> tuple[float, float, float]:
+        # Logic to handle location path
+
+        # This check is necessary
+        if address == "/location":
+            # Split the string argument into three float values
+            values = args[0].split(",")
+            try:
+                x, y, z = map(float, values)
+            except ValueError or len(x, y, z == 3):
+                return
+            self.values = x, y, z
+            return self.values
```

Comment:
```suggestion
            return x, y, z
```

Reply:
We need to assign self.values to something. Since the way wait_for_response only works when this is updated.

    def wait_for_response(self):
        """We wait for values to get assigned and then reset values to None for next check."""
        while not self.values:
            sleep(0.01)
        response = self.values
        self.values = None
        return response
        
If we simply return x,y,z instead of updating self.values it'll run for ever which is why me and Anjali added it back.